akka-sse/src/main/scala/de/heikoseeberger/akkasse/ServerSentEvent.scala

```diff
@@ -113,23 +133,39 @@ object ServerSentEvent {
  * @param id event id, must not contain \n or \r
  * @param retry the reconnection time in milliseconds.
  */
-final case class ServerSentEvent(data: String, eventType: Option[String] = None, id: Option[String] = None, retry: Option[Int] = None) {
+final case class ServerSentEvent(
+  data: String,
+  eventType: Option[String] = None,
+  id: Option[String] = None,
+  retry: Option[Int] = None
+)
+    extends EventStreamElement {
   import ServerSentEvent._
   require(eventType.forall(_.forall(c => c != '\n' && c != '\r')), "Event type must not contain \\n or \\r!")
   require(id.forall(_.forall(c => c != '\n' && c != '\r')), "Id must not contain \\n or \\r!")
   require(retry.forall(_ > 0L), "Retry must be a positive number!")
 
   /**
+   * Encodes this Server-Sent Event to an `akka.util.ByteString` according to the
+   * [[http://www.w3.org/TR/eventsource/#event-stream-interpretation SSE specification]].
+   *
+   * @return message converted to UTF-8 encoded `akka.util.ByteString`
+   */
+  override def encode = ByteString(toString, UTF_8.name)
```

Comment:
Sorry, I don't understand. Are you saying that first creating a `String` and later converting that into a `ByteString` is costlier than directly building `ByteString` fragments and concatenating these? I had that thought, but since you already went through the existing approach in depth (thereby rearranging my simple yet low performance FP code) I thought that this approach has the "approved by the Legend" stamp.